tools/run_nvfuser_tests.py

```diff
@@ -0,0 +1,567 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024-present NVIDIA CORPORATION & AFFILIATES.
+# All rights reserved.
+# SPDX-License-Identifier: BSD-3-Clause
+import subprocess
+import os
+import glob
+import datetime
+from pathlib import Path
+import time
+import argparse
+
+
+def setup_logging_dir():
+    """Create a timestamped directory for test logs and update latest symlink"""
+    timestamp = datetime.datetime.now().strftime("%Y%m%d_%H%M%S")
+    log_dir = Path(f"test_log_{timestamp}")
+    log_dir.mkdir(exist_ok=True)
+
+    # Check and handle the 'latest' symlink
+    latest_link = Path("test_log_latest")
+    if latest_link.exists():
+        if not latest_link.is_symlink():
+            raise RuntimeError(
+                "test_log_latest exists but is not a symlink. "
+                "Please remove it manually to proceed."
+            )
+        latest_link.unlink()  # Remove existing symlink
+
+    # Create new symlink pointing to the new log directory
+    latest_link.symlink_to(log_dir, target_is_directory=True)
+
+    return log_dir
+
+
+def get_cpp_test_executables(build_dir):
+    """Find all test executables in the build directory"""
+    all_tests = glob.glob(os.path.join(build_dir, "*test*"))
+
+    # Separate multidevice and single device tests
+    multidevice_tests, single_device_tests = [], []
+    for test in all_tests:
+        (single_device_tests, multidevice_tests)[
+            "multidevice" in os.path.basename(test).lower()
+        ].append(test)
+
+    # Separate priority tests and others
+    priority_tests = [
+        os.path.join(build_dir, "nvfuser_tests"),
+        os.path.join(build_dir, "test_matmul"),
+    ]
+    other_tests = [test for test in single_device_tests if test not in priority_tests]
+
+    # Return multidevice tests separately, and ordered single device tests prioritizing long running tests first
+    return multidevice_tests, priority_tests + other_tests
+
+
+def get_python_tests(python_test_dir):
+    """Find all Python test files"""
+    all_tests = glob.glob(os.path.join(python_test_dir, "test_*.py"))
+
+    # Separate multidevice and single device tests
+    multidevice_tests, single_device_tests = [], []
+    for test in all_tests:
+        (single_device_tests, multidevice_tests)[
+            "multidevice" in os.path.basename(test).lower()
```

Comment:
Can you add a code comment here that this misses multi-GPU tests in files like test_communication.py?